src/lib/dbManager.js

```diff
@@ -0,0 +1,364 @@
+/*
+  * Licensed Materials - Property of IBM
+  * (C) Copyright IBM Corp. 2016. All Rights Reserved.
+  * US Government Users Restricted Rights - Use, duplication or
+  * disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+  */
+'use strict';
+const path = require('path');
+const TAG = path.basename(__filename);
+const logger = require('./logger');
+const env = require('./env');
+const PouchDB = require('./PouchDB');
+
+const pjson = require(path.resolve(process.cwd(), 'package.json'));
+const classesDesignDoc = '_design/classes';
+const classesView = 'classes/byClass';
+const targetView = 'classes/byTarget';
+
+
+function DBManager(options){
+	this.localDbName = options.localDbName;
+	this.db = PouchDB.open(options.localDbName);
+	this.opts = options;
+
+	initializeDB(this.db).then(() => {
+		syncFn(this.db, options.localDbName, options.remoteDbName);
+	}).catch((error) => {
+		logger.error(`${TAG}: Unable to start sync for [${options.localDbName}] because`, error);
+	});
+
+}
+
+
+function initializeDB(db){
+	return new Promise((resolve, reject) => {
+		db.get('_design/classes').then(() => {
+			resolve();
+		}).catch(() => {
+			// create design doc
+			let ddoc = {
+				_id: classesDesignDoc,
+				storageType: 'private',
+				views: {
+					byClass: {
+						map: 'function(doc) { if (doc.selectedClass && doc.approved){ emit(doc.selectedClass); } if (doc.class && doc.text){ emit(doc.class); }}'
+					},
+					byTarget: {
+						map: 'function(doc){ if (doc.emittarget) { if(doc.description){ emit(doc._id, [doc.emittarget, doc.parameters, doc.description]); } else{ emit(doc._id, [doc.emittarget, doc.parameters]); }}}'
+					}
+				}
+			};
+
+			this.db.put(ddoc).then(() => {
```

Comment:
'this' keyword will not be what you expect from within the initializeDB helper function.  You should just do 'db.put(ddoc)...'.